proto/buf/alpha/registry/v1alpha1/sync.proto

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "buf/alpha/module/v1alpha1/module.proto";
+import "buf/alpha/registry/v1alpha1/git_metadata.proto";
+
+message GitSyncPoint {
+  string owner = 1;
+  string repository = 2;
+  string branch = 3;
+  string git_commit_hash = 4;
+  string bsr_commit_name = 5;
+}
+
+// SyncService is the Sync service.
+service SyncService {
+  // GetGitSyncPoint retrieves the Git sync point for the named repository
+  // on the specified branch.
+  rpc GetGitSyncPoint(GetGitSyncPointRequest) returns (GetGitSyncPointResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // SyncGitCommit syncs a Git commit containing a module to a named repository.
+  rpc SyncGitCommit(SyncGitCommitRequest) returns (SyncGitCommitResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+}
+
+message GetGitSyncPointRequest {
+  string owner = 1;
+  string repository = 2;
+  string branch = 3;
+}
+
+message GetGitSyncPointResponse {
+  GitSyncPoint sync_point = 1;
+}
+
+// SyncGitCommitRequest specifies the module to push to the BSR.
+message SyncGitCommitRequest {
+  string owner = 1;
+  string repository = 2;
+  string branch = 3;
+  // Manifest with all the module files being pushed.
+  buf.alpha.module.v1alpha1.Blob manifest = 4;
+  // Referenced blobs in the manifest. Keep in mind there is not necessarily one
+  // blob per file, but one blob per digest, so for files with exactly the same
+  // content, you can send just one blob.
+  repeated buf.alpha.module.v1alpha1.Blob blobs = 5;
```

Comment:
Don't quite understand why these are present here.

- Don't seem to be strictly part of any git metadata, it's what's pushed.
- If manifest and blobs matches with latest, it'll attach the git metadata to that latest BSR commit,
- If not, then it'll create a new commit (basically push effect)?

Reply:
Based on our conversation, I think we want a single push RPC that also handles the git metadata. So this will be called in lieu of `PushM+B` for sync.

Reply:
The current patterns we have around this are based on where we're coming from, not where we're going. Pretty soon we'll allow tags to move, so `push --tag <foo>`, even with identical content, should move the tag to the latest commit. Internally that may call `Latest` + either `Push` or `SetTag`, but that RPC layout is suboptimal.

I think where we're headed is a world in which everyone uses `buf sync` exclusively and `push --tag` becomes obsolete. `push` would remain for bespoke/advanced use cases where you need to fall down to the primitive, and this would be used in cases where you're not syncing something, per se (like dynamically downloaded and pushed schemas, somewhat like managed modules).